Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean

```diff
@@ -470,11 +539,12 @@ private lemma respects_isOpenImmersion_aux {X Y : Scheme.{u}} [IsAffine Y] {U :
     rw [← Category.assoc]
     exact this _ _ (IsLocalAtSource.comp hf _) V.2
   rw [HasRingHomProperty.iff_of_isAffine (P := P)] at hf ⊢
-  exact (isLocal_ringHomProperty P).StableUnderCompositionWithLocalizationAway.right _ a _ hf
+  exact hQ _ a _ hf
 
 /-- Any property of scheme morphisms induced by a property of ring homomorphisms is stable
 under composition with open immersions. -/
-instance respects_isOpenImmersion : P.Respects @IsOpenImmersion where
+lemma respects_isOpenImmersion (hQ : RingHom.StableUnderCompositionWithLocalizationAwayLeft Q) :
+    P.Respects @IsOpenImmersion where
```

Comment:
This is no longer an instance. Does this cause problems?

Reply:
I am a bit sad about it, but I still agree that weakening is good. Maybe we can add a `HasStrongRingHomProperty` extending `HasRingHomProperty` with the `RingHom.StableUnderCompositionWithLocalizationAwayLeft` as an extra assumption? Or maybe even call the new `PropertyIsLocal` `PropertyIsPreLocal` and let `PropertyIsLocal` extend `PropertyIsPreLocal` with the additional field `RingHom.StableUnderCompositionWithLocalizationAwayLeft`.

Reply:
Then we would have alignment with the stacks project definition https://stacks.math.columbia.edu/tag/01SR.

Reply:
Alternatively we could think about migrating to a type synonym `RingHomProperty` and then define all these properties as typeclasses.

Reply:
For this specific case it might be nice to establish `RingHom.StableUnderCompositionWithLocalizationAwayLeft` <=> `Respects IsOpenImmersion` but then I think we can just use `HasRingHomProperty` + `Respects IsOpenImmersion` later on.

In general it might be nice to migrate to typeclasses for these ringhom properties. It doesn't cause any immediate pain for me though so I probably won't work on this refactor for the near future.